### ISSUE TITLE
hvobj.opts.clear() of composite object clears opts of component objects but not combined object

### ISSUE THREAD

**Issue by JessicaS11** [holoviz/holoviews#4615]
## software version info
Python: 3.7.6
bokeh: 2.0.1
holoviews: 1.13.2
notebook: 6.0.3
numpy: 1.18.4
ipython: 7.15.0

## Expected vs Observed behavior
I have two holoviews image objects, img1 and img2, and create a composite object `imgs = img1+img2`. If I set opts parameters for img1 and img2 and do not override them with opts parameters in imgs, I would expect `imgs.opts.clear()` to only remove the opts settings specific to imgs. Instead, it leaves imgs.opts as is and clears img1.opts and img2.opts.

## MWE and screenshots
```
import holoviews as hv
hv.extension('bokeh','matplotlib')
from holoviews import dim, opts
import numpy as np

img1 = hv.Image((range(4), range(2), np.random.rand(2, 4)), datatype=['grid']).opts(cmap='reds')
img2 = hv.Image((range(4), range(2), np.random.rand(2, 4)), datatype=['grid']).opts(cmap='blues')
imgs = img1 + img2
imgs.opts(width=500)
imgs
```
<img width="720" alt="image" src="https://user-images.githubusercontent.com/11756442/93389006-8962cf80-f839-11ea-80ab-45cce1804497.png">

```
img1.opts.info()
img2.opts.info()
imgs.opts.info()
```

:Image   [x,y]   (z)
 | Options(cmap='reds')
:Image   [x,y]   (z)
 | Options(cmap='blues')
:Layout
 | Options(width=500)
   .Image.I  :Image   [x,y]   (z)
    | Options(cmap='reds')
   .Image.II :Image   [x,y]   (z)
    | Options(cmap='blues')

```
imgs.opts.clear()
```
<img width="720" alt="image" src="https://user-images.githubusercontent.com/11756442/93389118-b1523300-f839-11ea-83c9-036490d18c54.png">

```
img1.opts.info()
img2.opts.info()
imgs.opts.info()
```

:Image   [x,y]   (z)
:Image   [x,y]   (z)
:Layout
 | Options(width=500)
   .Image.I  :Image   [x,y]   (z)
   .Image.II :Image   [x,y]   (z)

After running clear on `imgs`, I'd expect this last output to be:
:Image   [x,y]   (z)
 | Options(cmap='reds')
:Image   [x,y]   (z)
 | Options(cmap='blues')
:Layout
   .Image.I  :Image   [x,y]   (z)
    | Options(cmap='reds')
   .Image.II :Image   [x,y]   (z)
    | Options(cmap='blues')

Which is the intended behavior?

**Comment by jlstevens** [comment]
Thanks for reporting this!

> Which is the intended behavior?

That is a good question as I see uses for both behaviors: clearing only the options on the container object or clearing the elements in the container. I think by default, clearing the container only makes sense (as you can index in and clear individual elements as necessary) but maybe a `recursive` flag (or similar) in `.clear` wouldn't go amiss.

I would need to check why and how the current behavior exists to be sure but this seems like a decent feature request.

**Comment by JessicaS11** [comment]
> I think by default, clearing the container only makes sense (as you can index in and clear individual elements as necessary) but maybe a `recursive` flag (or similar) in `.clear` wouldn't go amiss.

I agree completely. Acknowledging my elementary knowledge of holoviews, it would seem to me that currently there is no way to clear the options on the container object without deleting it and recreating it (even if you do `.clear` the options of its elements). I think a `recursive` flag is a great idea, though then we have the question of whether it deletes ONLY the options on all the elements or the options on all the elements AND the options on the container. Is there a `container.elements` type accessor that could be used with `.clear`, such that we'd have something like: `container.clear`, `container.elements.clear`, and `element.clear` to clear options (respectively) on just the container (the `recursive` flag could live here to specify just the container or the container and its elements), recursively on the elements in the container (but not on the container), and just on the element.
